### PR TITLE
added generator in metadata

### LINKED ISSUE #127: Querying metadata should return an iterator instead of a catalog
Like discussd in https://github.com/dlr-eoc/ukis-pysat/pull/99#discussion_r529627893_ I am not convinced of our current decision here. This might be okay if you are querying an AOI, but is very confusing for single `srcid`s. You are getting a catalog with only one item and if you want to query multiple `srcid`s you are receiving multiple catalogs as well. I agree however, that the return should be the same for both methods (`query_metadata()` & `query_metadata_srcid()`). An **iterator** would solve that in my opinion.

### PR DESCRIPTION
**Pull request by shiwakotisurendra**
closes #127

### CODE REVIEW

File: ukis_pysat/data.py [hunk 1]
```diff
@@ -130,11 +130,10 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
         :param date: Date from - to in format yyyyMMdd (String or Datetime tuple).
         :param aoi: Area of interest as GeoJson file or bounding box tuple with lat lon coordinates (String, Tuple).
         :param cloud_cover: Percent cloud cover scene from - to (Integer tuple).
-        :returns: Metadata catalog of products that match query criteria (PySTAC Catalog).
+        :generates: Metadata catalog of products that match query criteria (PySTAC Catalog).
```
Comment (fwfichtner) [inline ukis_pysat/data.py:133]: it just generates the item now, right?

File: ukis_pysat/data.py [hunk 2]
```diff
@@ -184,28 +182,23 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
...
             products = self.api.to_geojson(products)["features"]
 
-        # initialize empty catalog and add metadata items
-        catalog = self._init_catalog()
         for meta in products:
-            catalog.add_item(self.construct_metadata(meta=meta, platform=platform))
+            yield self.construct_metadata(meta=meta, platform=platform)
 
-        return catalog
+        # return catalog
```
Comment (fwfichtner) [inline ukis_pysat/data.py:188]: better delete it right away please

File: ukis_pysat/data.py [hunk 3]
```diff
@@ -184,28 +182,23 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
...
 
     def query_metadata_srcid(self, platform, srcid):
         """Queries metadata from data source by srcid.
 
         :param platform: Image platform (<enum 'Platform'>).
         :param srcid: Srcid of a specific product (String).
-        :returns: Metadata of product that matches srcid (PySTAC Catalog).
+        :generates: Metadata of product that matches srcid (PySTAC Catalog).
```
Comment (fwfichtner) [inline ukis_pysat/data.py:195]: it is also just the item here

**Review by fwfichtner — CHANGES_REQUESTED** [review]
Looks good, that's what we had in mind. Just some minor adaptions and then the tests.

**Review by fwfichtner — APPROVED** [review]
well done, @MWieland is this good for you as well?

File: CHANGELOG.rst [hunk 4]
```diff
@@ -18,14 +18,14 @@ something was changed
 
 Changed
 ^^^^^^^
+- ``data``: generator added to query_metadata and query_metadata_srcid #127
```
Comment (fwfichtner) [inline CHANGELOG.rst:21]:
```suggestion
- ``data``: generator instead of catalog return for query_metadata and query_metadata_srcid #127
```

**Review by fwfichtner — APPROVED** [review]

**Review by MWieland — APPROVED** [review]
Looks good to me. Well done.

**Pull request merged**